aggregator/src/diagnostic.rs

```diff
@@ -0,0 +1,252 @@
+//! Writing diagnostic files to disk.
+
+use anyhow::Context;
+use derivative::Derivative;
+use janus_messages::{AggregationJobId, ReportMetadata, TaskId};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{fs::File, path::Path, time::SystemTime};
+use uuid::Uuid;
+
+/// Represents an illegal attempt to mutate an aggregation job.
+#[derive(Derivative, Clone, Serialize, Deserialize)]
+#[derivative(Debug)]
+pub struct AggregationJobInitForbiddenMutationEvent {
+    /// The ID of the task.
+    #[serde(with = "serialize_task_id")]
+    pub task_id: TaskId,
+
+    /// The ID of the aggregation job.
+    #[serde(with = "serialize_job_id")]
```

Comment:
Yeah, and it's worth my while to derive `Deserialize` so I can use it in the unit test to check the contents of the written event, since I use a different type as the inner event there. Additionally, I suspect we will soon add real implementations of `Deserialize` on the illegal mutation event so we can write tools that ingest the JSON events and analyze them, so I don't feel _too_ bad about the TODOs.